indico/modules/designer/pdf.py

```diff
@@ -92,22 +94,34 @@ def get_pdf(self):
         return data
 
     def _draw_item(self, canvas, item, tpl_data, content, margin_x, margin_y):
-        style = ParagraphStyle({})
-        style.alignment = ALIGNMENTS[item['text_align']]
-        style.textColor = item.get('color') or '#000000'
-        style.backColor = item.get('background_color') or None
-        style.borderPadding = (0, 0, 4, 0)
-        style.fontSize = _extract_font_size(item['font_size'])
-        style.leading = style.fontSize
+        font_size = _extract_font_size(item['font_size'])
+        styles = {
+            'alignment': ALIGNMENTS[item['text_align']],
+            'textColor': item.get('color') or '#000000',
+            'backColor': item.get('background_color') or None,
+            'borderPadding': (0, 0, 4, 0),
+            'fontSize': font_size,
+            'leading': font_size
+        }
 
         if item['bold'] and item['italic']:
-            style.fontName = FONT_STYLES[item['font_family']][3]
+            styles['fontName'] = FONT_STYLES[item['font_family']][3]
         elif item['italic']:
-            style.fontName = FONT_STYLES[item['font_family']][2]
+            styles['fontName'] = FONT_STYLES[item['font_family']][2]
         elif item['bold']:
-            style.fontName = FONT_STYLES[item['font_family']][1]
+            styles['fontName'] = FONT_STYLES[item['font_family']][1]
         else:
-            style.fontName = FONT_STYLES[item['font_family']][0]
+            styles['fontName'] = FONT_STYLES[item['font_family']][0]
+
+        for update in values_from_signal(
+            signals.event.designer.update_badge_style.send(item, styles=styles),
```

Comment:
Yes, good idea. Although neither the template nor the registration is available in this function, would it be a okay to use the `tpl_data` as the sender?

Reply:
considering that this is not useful to match e.g. on the event or template used (check the data in there...) I don't think so. But I don't see any reason against passing this data. Just make sure posters still work since those use this method as well but obviously don't have a registration associated